src/rendering/batcher/shared/Batcher.ts

```diff
@@ -65,7 +65,7 @@ export interface BatchableObject
         index: number,
         textureId: number,
     ) => void;
-    packIndex: (indexBuffer: Uint32Array, index: number, indicesOffset: number) => void;
+    packIndex: (indexBuffer: Uint32Array | Uint16Array, index: number, indicesOffset: number) => void;
```

Comment:
Maybe we should make this a specific type and use instead of the `|` signature. There seem to be enough of them to warrant it.

For example:
```
type IndexBufferArray = Uint32Array | Uint16Array;

packIndex: (indexBuffer: IndexBufferArray, index: number, indicesOffset: number) => void;
```
```

Reply:
good shout! PR updated